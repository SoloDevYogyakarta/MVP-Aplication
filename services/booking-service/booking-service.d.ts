import { HttpStatus } from '@nestjs/common';
import { BookingField } from '../../dto/booking-dto/booking-dto';
export declare class BookingService {
    private readonly logger;
    create(body: BookingField): Promise<{
        status: HttpStatus;
        message: string;
        create: import("../../database/entities/services/booking-entity/booking-entity").BookingInstance;
    }>;
    destroy(id: number): Promise<{
        status: HttpStatus;
        message: string;
    }>;
}
