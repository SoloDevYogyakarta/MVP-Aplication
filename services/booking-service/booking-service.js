"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var BookingService_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.BookingService = void 0;
const common_1 = require("@nestjs/common");
const user_entity_1 = require("../../database/entities/authenticate/user-entity/user-entity");
const booking_entity_1 = require("../../database/entities/services/booking-entity/booking-entity");
let BookingService = BookingService_1 = class BookingService {
    constructor() {
        this.logger = new common_1.Logger(BookingService_1.name);
    }
    async create(body) {
        this.logger.log(BookingService_1.name);
        const findOne = await user_entity_1.userEntity.findOne({
            where: { id: body?.user_id ?? 0 },
        });
        if (findOne) {
            body = { ...body, user_id: findOne.id };
        }
        const create = await booking_entity_1.bookingEntity.create({ ...body });
        create.save();
        return {
            status: common_1.HttpStatus.CREATED,
            message: 'Booking has been create',
            create,
        };
    }
    async destroy(id) {
        const findOne = await booking_entity_1.bookingEntity.findOne({ where: { id } });
        if (!findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: 'Booking not found',
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        findOne.destroy();
        return { status: common_1.HttpStatus.OK, message: 'Booking has been delete' };
    }
};
exports.BookingService = BookingService;
exports.BookingService = BookingService = BookingService_1 = __decorate([
    (0, common_1.Injectable)()
], BookingService);
//# sourceMappingURL=booking-service.js.map