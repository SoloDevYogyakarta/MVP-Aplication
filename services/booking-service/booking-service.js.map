{"version":3,"file":"booking-service.js","sourceRoot":"","sources":["../../../src/services/booking-service/booking-service.ts"],"names":[],"mappings":";;;;;;;;;;AAAA,2CAA+E;AAC/E,8FAA0F;AAC1F,mGAA+F;AAIxF,IAAM,cAAc,sBAApB,MAAM,cAAc;IAApB;QACY,WAAM,GAAG,IAAI,eAAM,CAAC,gBAAc,CAAC,IAAI,CAAC,CAAC;IAiC5D,CAAC;IA/BC,KAAK,CAAC,MAAM,CAAC,IAAkB;QAC7B,IAAI,CAAC,MAAM,CAAC,GAAG,CAAC,gBAAc,CAAC,IAAI,CAAC,CAAC;QACrC,MAAM,OAAO,GAAG,MAAM,wBAAU,CAAC,OAAO,CAAC;YACvC,KAAK,EAAE,EAAE,EAAE,EAAE,IAAI,EAAE,OAAO,IAAI,CAAC,EAAE;SAClC,CAAC,CAAC;QACH,IAAI,OAAO,EAAE;YACX,IAAI,GAAG,EAAE,GAAG,IAAI,EAAE,OAAO,EAAE,OAAO,CAAC,EAAE,EAAE,CAAC;SACzC;QACD,MAAM,MAAM,GAAG,MAAM,8BAAa,CAAC,MAAM,CAAC,EAAE,GAAG,IAAI,EAAE,CAAC,CAAC;QACvD,MAAM,CAAC,IAAI,EAAE,CAAC;QACd,OAAO;YACL,MAAM,EAAE,mBAAU,CAAC,OAAO;YAC1B,OAAO,EAAE,yBAAyB;YAClC,MAAM;SACP,CAAC;IACJ,CAAC;IAED,KAAK,CAAC,OAAO,CAAC,EAAU;QACtB,MAAM,OAAO,GAAG,MAAM,8BAAa,CAAC,OAAO,CAAC,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAAC,CAAC;QAC/D,IAAI,CAAC,OAAO,EAAE;YACZ,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,mBAAmB;aAC7B,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QACD,OAAO,CAAC,OAAO,EAAE,CAAC;QAClB,OAAO,EAAE,MAAM,EAAE,mBAAU,CAAC,EAAE,EAAE,OAAO,EAAE,yBAAyB,EAAE,CAAC;IACvE,CAAC;CACF,CAAA;AAlCY,wCAAc;yBAAd,cAAc;IAD1B,IAAA,mBAAU,GAAE;GACA,cAAc,CAkC1B"}