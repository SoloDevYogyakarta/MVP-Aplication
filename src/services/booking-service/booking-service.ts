import { HttpException, HttpStatus, Injectable, Logger } from '@nestjs/common';
import { userEntity } from '../../database/entities/authenticate/user-entity/user-entity';
import { bookingEntity } from '../../database/entities/services/booking-entity/booking-entity';
import { BookingField } from '../../dto/booking-dto/booking-dto';

@Injectable()
export class BookingService {
  private readonly logger = new Logger(BookingService.name);

  async create(body: BookingField) {
    this.logger.log(BookingService.name);
    const findOne = await userEntity.findOne({
      where: { id: body?.user_id ?? 0 },
    });
    if (findOne) {
      body = { ...body, user_id: findOne.id };
    }
    const create = await bookingEntity.create({ ...body });
    create.save();
    return {
      status: HttpStatus.CREATED,
      message: 'Booking has been create',
      create,
    };
  }

  async destroy(id: number) {
    const findOne = await bookingEntity.findOne({ where: { id } });
    if (!findOne) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: 'Booking not found',
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    findOne.destroy();
    return { status: HttpStatus.OK, message: 'Booking has been delete' };
  }
}
