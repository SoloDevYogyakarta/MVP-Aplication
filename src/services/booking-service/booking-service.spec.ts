import { BookingService } from './booking-service';

describe('BookingService', () => {
  let service: BookingService;

  beforeEach(() => {
    service = new BookingService();
  });

  it('should to be defined', () => expect(service).toBeDefined());

  it('render correctly', () => expect(service).toMatchSnapshot());
});
