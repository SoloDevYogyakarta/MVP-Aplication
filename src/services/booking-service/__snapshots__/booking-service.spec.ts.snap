// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingService render correctly 1`] = `
BookingService {
  "logger": Logger {
    "context": "BookingService",
    "options": {},
  },
}
`;
